contracts/VaultV3.vy

```diff
@@ -551,13 +551,88 @@ def _max_deposit(receiver: address) -> uint256:
+    not account for it in the amount returned since it is unrealised and 
+    therefore will not be accounted for in the conversion rates.
+
+    i.e. If we have 100 debt and 10 of unrealised loss, the max we can get
+    out is 90, but a user of the vault will need to call withdraw with 100
+    in order to get the full 90 out.
+    """
+    # Get the max amount for the owner if fully liquid.
+    max_assets: uint256 = self._convert_to_assets(self.balance_of[owner], Rounding.ROUND_DOWN)
+    
+    # See if we have enough idle to service the withdraw.
+    current_idle: uint256 = self.total_idle
+    if max_assets > current_idle:
+        # Amount left that we need.
+        needed: uint256 = unsafe_sub(max_assets, current_idle)
+        # Track how much we can pull.
+        have: uint256 = current_idle
+        loss: uint256 = 0
+
+        # If no queue was passed use the default one.
+        _strategies: DynArray[address, MAX_QUEUE] = strategies
+        if len(_strategies) == 0:
+            _strategies = self.default_queue
+
+        for strategy in _strategies:
+            # Can't use an invalid strategy.
+            assert self.strategies[strategy].activation != 0, "inactive strategy"
+
+            # Get the maximum amount the vault can withdraw from the strategy.
+            max_withdraw: uint256 = min(needed, self.strategies[strategy].current_debt)
+
+            # Get any unrealised loss for the strategy.
+            unrealised_loss: uint256 = self._assess_share_of_unrealised_losses(strategy, max_withdraw)
+
+            # See if any limit is enforced by the strategy.
+            strategy_limit: uint256 = IStrategy(strategy).maxWithdraw(self)
+
+            # Adjust accordingly if there is a max withdraw limit.
+            if strategy_limit < max_withdraw - unrealised_loss:
+                # lower unrealised loss to the proportion to the limit.
+                unrealised_loss = unrealised_loss * strategy_limit / max_withdraw
+                # Still count the unrealised loss as withdrawable.
+                max_withdraw = strategy_limit + unrealised_loss
+
+            # If 0 move on to the next strategy.
+            if max_withdraw == 0:
+                continue
+
+            # Add to what we can pull.
+            have += max_withdraw
+            # Reduce how much we have left.
+            needed -= max_withdraw
+            # Add any unrealised loss to the total
+            loss += unrealised_loss
+
+            if needed == 0:
+                break
+
+        # Update the max after going through the queue.
+        max_assets = have
+
+        # Check if there is a loss and a non-default value was set.
+        if loss > 0 and max_loss < MAX_BPS:
+            # If the loss is not within the allowed range.
+            if loss > max_assets * max_loss / MAX_BPS:
+                # The tx will revert. Can only withdraw idle.
+                max_assets = current_idle
+
+    return max_assets
```

Comment:
This is underestimates the max_assets available in cases where partially iterating through the withdraw queue would have had tolerable losses.

Reply:
https://github.com/yearn/yearn-vaults-v3/pull/179/commits/abb73b675c496a293a1562e792f1748aa011bf18

Added to the other branch with the modules.

I Did not account for a scenario in which partial would revert but the full queue would not. Think that we shouldnt return a value that has amounts less than it that would revert. Could break things.

Perhaps that could be something to add to a preivewWithdraw/redeem